nativeauthenticator/handlers.py

```diff
@@ -188,35 +224,45 @@ async def get(self):
         self.finish(html)
 
 
-class ChangeAuthorizationHandler(LocalBase):
+class ToggleAuthorizationHandler(LocalBase):
+    """Responsible for the authorize/[someusername] page,
+    which immediately redirects after toggling the
+    respective user's authorization status."""
+
     @admin_users_scope
     async def get(self, slug):
         UserInfo.change_authorization(self.db, slug)
         self.redirect(self.hub.base_url + "authorize#" + slug)
 
 
-class AuthorizeHandler(LocalBase):
+class EmailAuthorizationHandler(LocalBase):
+    """Responsible for the confirm/[someusername] validation of
+    cryptographic URLs for the self-serve-approval feature."""
+
     async def get(self, slug):
-        must_stop = True
+        """Called on GET requests. The slug is given in the URL."""
+
+        slug_validation_successful = False
         message = "Invalid URL"
+
         if self.authenticator.allow_self_approval_for:
             try:
-                data = AuthorizeHandler.validate_slug(
+                data = EmailAuthorizationHandler.validate_slug(
                     slug, self.authenticator.secret_key
                 )
-                must_stop = False
+                slug_validation_successful = True
             except ValueError:
                 pass
 
-        if not must_stop:
+        if slug_validation_successful:
             username = data["username"]
-            message = f"{username} was already authorized"
             usr = UserInfo.find(self.db, username)
+
             if not usr.is_authorized:
                 UserInfo.change_authorization(self.db, username)
-                message = f"{username} has been authorized"
-
-            # add POSIX user!!
```

Comment:
+1 to removing that comment. I think it's incorrect to add a POSIX user as a FIXME kind of comment, perhaps its relevant to add some support like this, but its a big deal to be covered in in depth in a dedicated issue/pr.